activerecord/lib/active_record/errors.rb

```diff
@@ -7,9 +7,9 @@ module ActiveRecord
   class ActiveRecordError < StandardError
   end
 
-  # Raised when trying to use a feature in Active Record which requires Active Job but the gem is not present.
-  class ActiveJobRequiredError < ActiveRecordError
-  end
+  # DEPRECATED: Previously raised when trying to use a feature in Active Record which
+  # requires Active Job but the gem is not present. Now raises a NameError.
+  ActiveJobRequiredError = ActiveSupport::Deprecation::DeprecatedConstantProxy.new("ActiveRecord::ActiveJobRequiredError", "NameError")
```

Comment:
Using a `ActiveSupport::Deprecation::DeprecatedConstantProxy` will display a deprecation warning only when a method is called on the proxy (e.g. when it's instantiated), rather than when it is referenced (e.g. in a `rescue` that is never triggered).

Also, although Active Job being absent will now raise a `NameError`, I think using `NameError` here could have unintended consequences.  It will cause any `rescue ActiveRecord::ActiveJobRequiredError` statements to rescue unrelated `NameError`s as well.

There probably aren't that many `rescue ActiveRecord::ActiveJobRequiredError` statements in the wild, so I think it might be okay to simply display a deprecation warning while rendering them inert.  Such as with:

```suggestion
  include ActiveSupport::Deprecation::DeprecatedConstantAccessor
  DeprecatedActiveJobRequiredError = Class.new(ActiveRecordError)
  deprecate_constant "ActiveJobRequiredError", "ActiveRecord::DeprecatedActiveJobRequiredError",
    message: "ActiveRecord::ActiveJobRequiredError has been deprecated. If Active Job is not present, a NameError will be raised instead."
```